src/query/pipeline/sources/src/input_formats/input_format_text.rs

```diff
@@ -289,8 +293,25 @@ impl<T: InputFormatTextBase> InputFormat for T {
                 stage_info.file_format_params.compression(),
                 &path,
             )?;
-            let split_size = stage_info.copy_options.split_size;
-            if compress_alg.is_none() && T::is_splittable() && split_size > 0 {
+
+            let mut split_size = stage_info.copy_options.split_size;
+            let mut splittable = is_splittable(&info);
+
+            if splittable {
+                // user doesn't specify the split
+                if split_size == 0 {
+                    split_size = TEXT_FILE_SPLIT_CHUNK_SIZE;
```

Comment:
should retain the ability to set split_size to 0 (disable it)
I prefer to change copy_options.split_size to some number when ready, no need to add const?